firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/OrderDirectionIntegrationTest.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.dataconnect
+
+import com.google.firebase.dataconnect.testutil.DataConnectIntegrationTestBase
+import com.google.firebase.dataconnect.testutil.sortedParallelTo
+import com.google.firebase.dataconnect.testutil.tag
+import io.kotest.common.DelicateKotest
+import io.kotest.matchers.collections.shouldContainExactly
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.property.Arb
+import io.kotest.property.arbitrary.distinct
+import io.kotest.property.arbitrary.int
+import io.kotest.property.arbitrary.next
+import kotlinx.coroutines.test.runTest
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.serializer
+import org.junit.Test
+
+class OrderDirectionIntegrationTest : DataConnectIntegrationTestBase() {
+
+  private val dataConnect: FirebaseDataConnect by lazy {
+    val connectorConfig = testConnectorConfig.copy(connector = "alltypes")
+    dataConnectFactory.newInstance(connectorConfig)
+  }
+
+  @OptIn(DelicateKotest::class) private val uniqueInts = Arb.int().distinct()
+
+  @Test
+  fun orderDirectionQueryVariableOmittedShouldUseAscendingOrder() = runTest {
+    val tag = Arb.tag().next(rs)
+    val values = List(5) { uniqueInts.next(rs) }
+    val insertedIds = insertRow(tag, values)
+
+    val queryIds = getRowIds(tag)
+
+    queryIds shouldContainExactlyInAnyOrder insertedIds
```

Comment:
Should this be in any order? I feel like the order should matter here

Reply:
Good question. My testing showed that it returned them in the same order they were inserted, but I'm guessing that's just coincidence and an implementation detail of postgresql. I'll ask.

Reply:
Asked here: https://chat.google.com/room/AAAAdvEjzno/lXUw2WzUwdo

Reply:
(answer: nope, its explicitly unspecified and shouldn't be relied on https://www.postgresql.org/docs/current/queries-order.html)